pkg/selinux/selinux-client.js

```diff
@@ -96,6 +102,66 @@ export function init(statusChangedCallback) {
         );
     };
 
+    var parseBoolean = function(result, item) {
+        // Example:
+        // authlogin_nsswitch_use_ldap (on , on) Allow authlogin to nsswitch use ldap
+        // Split by first ')', as the name cannot contain ')'
+        result.push(item.split(")", 2)[1].trim());
+        return result;
+    };
+
+    var getModifications = function() {
+        // List of items we know how to parse
+        let manageditems_callbacks = [["boolean", parseBoolean]];
+        let manageditems = manageditems_callbacks.map(item => item[0]);
+
+        // Building a query to get information from semanage
+        // Use `semanage export` to show shell script (and parse types we yet don't parse explicitly)
+        // Use `semanage <type> --list -C` to get better readable and parsable local changes
+        // Use `echo '~~~~~'` as separator, so we don't need to execute multiple commands
+        let script = "semanage export";
+        manageditems.forEach(item => { script += " && echo '~~~~~' && semanage " + item + " --list -C --noheading" });
+        cockpit.spawn(["/bin/sh", "-c", script], { err: 'message', environ: [ "LC_ALL=C" ], superuser: "try" }).then(
+            function(output) {
+                output = output.split("~~~~~");
+                status.shell = output[0];
+                status.modifications = [];
+
+                for (let i = 1; i < output.length; i++)
+                    status.modifications.push(...(output[i].trim().split("\n")
+                            .reduce(manageditems_callbacks[i - 1][1], [])));
+
+                // As long as we don't parse all items, we need to get some from general export
+                // Once we can parse all types, this can be dropped
+                status.modifications.push(...(output[0].split("\n").reduce(function (result, mod) {
+                    mod = mod.trim();
+                    if (mod === "")
+                        return result;
+
+                    let items = mod.split(" ");
+
+                    // Skip enumeration of types, e.g. 'boolean -D'
+                    if (items.length === 2 && items[1] == "-D")
+                        return result;
+
+                    if (manageditems.indexOf(items[0]) < 0)
+                        result.push(mod);
+                    return result;
+                }, [])));
+
+                statusChangedCallback(status, undefined);
+            }
+        )
+                .catch(e => {
+                    if (e.message === "ValueError: SELinux policy is not managed or store cannot be accessed.") {
```

Comment:
This is an awfully long and specific string to rely on. Would the `ValueError` be enough?

Reply:
I am not sure about this - I don't know what other possible errors can occur. But surely we can go with `ValueError` and then just see, if it may be misleading.